File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Jre5/Collections/Typehandler/AbstractItemFactory.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using System;

namespace Db4objects.Db4o.Tests.Jre5.Collections.Typehandler
{
	public abstract class AbstractItemFactory
	{
		internal static readonly string MapFieldName = "_map";

		internal static readonly string ListFieldName = "_list";

		public abstract string FieldName();

		public abstract object NewItem();

		public abstract Type ItemClass();

		public abstract Type ContainerClass();
	}
}

File: db4oj.tests/src/com/db4o/db4ounit/jre5/collections/typehandler/AbstractItemFactory.java
/* Copyright (C) 2008   Versant Inc.   http://www.db4o.com */

package com.db4o.db4ounit.jre5.collections.typehandler;

/**
 */
@decaf.Ignore(unlessCompatible=decaf.Platform.JDK15)
public abstract class AbstractItemFactory {
	
	static final String MAP_FIELD_NAME = "_map";
	static final String LIST_FIELD_NAME = "_list";
	
	public abstract String fieldName();
	public abstract Object newItem();
	public abstract Class itemClass();
	public abstract Class containerClass();
}
